src/test/java/org/takes/rq/RqMultipartTest.java

```diff
@@ -271,6 +272,100 @@ public void parsesHttpBody() throws IOException {
     }
 
     /**
+     * RqMultipart.Base can close all parts once the request body has been
+     * closed.
+     * @throws IOException If some problem inside
+     */
+    @Test(expected = IOException.class)
+    public void closesAllParts() throws IOException {
+        final String body = "RqMultipartTest.closesAllParts";
+        final RqMultipart request = new RqMultipart.Fake(
+            new RqFake(),
+            new RqGreedy(
+                new RqWithHeaders(
+                    new RqFake("", "", body),
+                    RqMultipartTest.contentLengthHeader(
+                        (long) body.getBytes().length
+                    ),
+                    RqMultipartTest.contentDispositionHeader(
+                        "form-data; name=\"name\""
+                    )
+                )
+            ),
+            new RqGreedy(
+                new RqWithHeaders(
+                    new RqFake("", "", body),
+                    RqMultipartTest.contentLengthHeader(0L),
+                    RqMultipartTest.contentDispositionHeader(
+                        "form-data; name=\"content\"; filename=\"a.bin\""
+                    )
+                )
+            )
+        );
+        final RqMultipart.Base multi = new RqMultipart.Base(request);
+        multi.body().close();
+        MatcherAssert.assertThat(
+            multi.part("name").iterator().next(),
+            Matchers.notNullValue()
+        );
+        try {
+            multi.part("name").iterator().next().body().read();
+            Assert.fail("An IOException was expected");
+        } catch (final IOException ex) {
+            MatcherAssert.assertThat(
+                multi.part("content").iterator().next(),
+                Matchers.notNullValue()
+            );
+            multi.part("content").iterator().next().body().read();
+        }
+    }
+
+    /**
+     * RqMultipart.Base can close all parts explicitly even if the request body
+     * has been closed.
```

Comment:
@mkordas I don't have access to the underlying Stream so the only way I have to ensure that it is closed is to try to read the stream and expect an IOException